File: Service/Translators/ClientOwnerTypeTranslator.vb
Imports DAL
Imports BO
Imports System.Text.RegularExpressions
Public Class ClientOwnerTypeTranslator

    Friend Shared Function TranslateEntityToBO(_entity As ClientOwnerType, bo As ClientOwnerTypeBO) As ErrorCode
        If _entity Is Nothing Then Return ErrorCode.EntityNull
        If bo Is Nothing Then Return ErrorCode.BoNull
        bo.Id = _entity.ID
        bo.Name = _entity.Name
        Return ErrorCode.Success
    End Function

    Friend Shared Function TranslateBOToEntity(_entity As ClientOwnerType, bo As ClientOwnerTypeBO, uow As UnitOfWork) As ErrorCode
        If _entity Is Nothing Then Return ErrorCode.EntityNull
        If bo Is Nothing Then Return ErrorCode.BoNull
        _entity.Name = bo.Name
        Return ErrorCode.Success
    End Function


End Class

File: ServiceCore/Translators/ClientOwnerTypeTranslator.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using BOCore;
using DALCore.Models;
using DALCore;

namespace ServiceCore.Translators
{
    class ClientOwnerTypeTranslator
    {
        internal static ErrorCode TranslateEntityToBO(ClientOwnerType _entity, ClientOwnerTypeBO bo)
        {
            if (_entity == null)
                return ErrorCode.EntityNull;
            if (bo == null)
                return ErrorCode.BoNull;
            bo.Id = _entity.Id;
            bo.Name = _entity.Name;
            return ErrorCode.Success;
        }

        internal static ErrorCode TranslateBOToEntity(ClientOwnerType _entity, ClientOwnerTypeBO bo, UnitOfWork uow)
        {
            if (_entity == null)
                return ErrorCode.EntityNull;
            if (bo == null)
                return ErrorCode.BoNull;
            _entity.Name = bo.Name;
            return ErrorCode.Success;
        }
    }
}
